src/groups/metric_group.jl

```diff
@@ -0,0 +1,100 @@
+"""
+    MetricLieGroup{𝔽,M<:AbstractLieGroup{𝔽},G<:AbstractMetric} <: AbstractDecoratorManifold{𝔽}
+
+Equip an [`AbstractLieGroup`](@ref) `L` explicitly with an [`AbstractMetric`](@extref `ManifoldsBase.AbstractMetric`) `G`.
+
+By default every [`AbstractLieGroup`](@ref) `L` is assumed to implicitly
+implement all functions on their [`LieAlgebra`](@ref) with respect to a metric,
+that corresponds to a certain “default metric”, either because it is a widely recognized metric
+or because the first implementation within `LieGroups.jl` was done with respect to this metric.
+
+This default is usually indicated by checking that [`metric(L)`](@extref `Manifolds`) returns [`DefaultMetric`](@extref `Manifolds.DefaultMetric`).
+
+This decorator type allows to explicitly equip the Lie group with a different metric.
+Note that for any functions unrelated to the metric, this new Lie group will simply forward
+to the Lie group that is internally stored.
+This for example includes group operations like [`compose`], `inv`, or `identity_element`,
+or functions that are already otherwise passed on to the inner manifold like [`manifold_dimension`](@extref `ManifoldsBase.manifold_dimension-Tuple{AbstractManifold}`).
+
+# Constructor
+
+    MetricManifold(M, G)
+
+Generate the [`AbstractLieGroup`](https://juliamanifolds.github.io/Manifolds.jl/latest/interface.html#ManifoldsBase.AbstractLieGroup) `M` as a manifold with the `AbstractMetric` `G`.
+"""
+struct MetricLieGroup{
+        𝔽,
+        O <: AbstractGroupOperation,
+        M <: ManifoldsBase.AbstractManifold{𝔽},
+        L <: LieGroup{𝔽, O, M},
+        G <: AbstractMetric,
+    } <: AbstractLieGroup{𝔽, O, M}
+    lie_group::L
+    metric::G
+end
+
+Base.getindex(g, G::MetricLieGroup, i) = getindex(g, base_lie_group(G), i)
+Base.getindex(g::AbstractArray, G::MetricLieGroup, i) = getindex(g, base_lie_group(G), i)
+function Base.getindex(
+        X, 𝔤::LieAlgebra{𝔽, <:O, <:MetricLieGroup}, i
```

Comment:
```suggestion
        X, 𝔤::LieAlgebra{𝔽, O, <:MetricLieGroup}, i
```
`O` is upper-bounded in the where clause, no need to `<:` here.